.gitlab/docker/Dockerfile.spack_compiler

```diff
@@ -25,7 +25,7 @@ RUN echo "spack arch: $(spack arch)"
 
 # Install dependencies for this configuration
 ARG SPACK_SPEC
+ARG DEPRECATED=
```

Comment:
Is it clearer to maybe call this `ALLOW_DEPRECATED` and explicitly set it to `yes`/`no` or `true`/`false`? (and default it to false)